### PR TITLE
Fixing support for mpirun (and similar)

### PR DESCRIPTION
**Pull request by hkaiser**
- Checking MPI environment before initializing the runtime
- flyby: handling stack size problem for newly created threads

### CODE REVIEW

File: libs/threading_base/include/hpx/threading_base/scheduler_base.hpp [hunk 1]
```diff
@@ -251,8 +251,9 @@ namespace hpx { namespace threads { namespace policies {
             switch (stacksize)
             {
             case thread_stacksize_small:
+                HPX_FALLTHROUGH;
+            case thread_stacksize_current:
```
Comment (msimberg) [inline libs/threading_base/include/hpx/threading_base/scheduler_base.hpp:255]: Did you have a case where `stacksize` still was `thread_stacksize_current` at this point? I would rather put an assert in the first if after `stacksize` has been updated as this shouldn't happen. A created HPX thread should always have one of the actual sizes (small, medium, ...), and a non-HPX thread should return the default which is the small stacksize at the moment.

Comment (hkaiser) [reply libs/threading_base/include/hpx/threading_base/scheduler_base.hpp:255]: Yes, that's what I was seeing. I did add an assert inside the `if()`, which fired for me. I tried to investigate but was not able to figure out why this happens.

Comment (msimberg) [reply libs/threading_base/include/hpx/threading_base/scheduler_base.hpp:255]: Hrm, I'm clearly wrong in that case. I think what we should do is reset a `thread_stacksize_current` to the a more concrete size when creating the thread. This is most likely a bug from my earlier changes. We can go with this and I'll try to fix it afterwards.

**Comment by hkaiser** [comment]
@khuck wrote in an email (Tue 5/19/2020 5:55 PM): 
> I can confirm that the branch works for me…thanks!  I am able to run MPI programs again.

File: libs/threading_base/include/hpx/threading_base/scheduler_base.hpp (hunk 1 above)
Comment (hkaiser) [reply libs/threading_base/include/hpx/threading_base/scheduler_base.hpp:255]: Yes, I don't like this 'fix' either. See #4648 for a separate ticket.

**Comment by msimberg** [comment]
Thanks @hkaiser for opening the new issue. I'll take care of that one. Let's go with this since having the MPI parcelport broken is suboptimal...

**Pull request merged**